integration/jnifuse/fs/src/main/java/alluxio/jnifuse/AbstractFuseFileSystem.java

```diff
@@ -189,11 +189,18 @@ public int openCallback(String path, ByteBuffer buf) {
     }
   }
 
-  public int readCallback(String path, ByteBuffer buf, long size, long offset, ByteBuffer fibuf) {
+  public int readCallback(String path, ByteBuffer buf, long _size, long position,
+      ByteBuffer fibuf) {
+    // this buffer is allocated by libfuse and wrapped as a DirectByteBuffer,
+    // its entirety should be appropriate for writing, so set its position to the beginning
```

Comment:
Have you check the libfuse and JNIFUSE code to make sure it's correct?
If it's this case, they do not need to provide the size information

Reply:
The buffer is created in our JNI code: https://github.com/Alluxio/alluxio/blob/46271636318e22eeb499291e68529f78743292e4/integration/jnifuse/native/src/main/native/libjnifuse/operation.cc#L111
The size argument from libfuse is used to specify the capacity of the buffer. The JNI documentation says nothing about the limit of the buffer after its creation, so I added a `buf.clear()` to make sure the limit is set to the capacity.

The actual memory behind the buffer is allocated by libfuse before calling our callback:
https://github.com/libfuse/libfuse/blob/d8d1f84200ca559b202d331ebe001224c2de9be5/lib/fuse.c#L1766-L1775
The buffer is guaranteed to be of size `size` and appropriate for writing.